src/Infrastructure/FunctionControllerBase.cs

```diff
@@ -165,33 +166,32 @@ protected virtual bool TryHandleError(Exception ex, out IActionResult response)
             return true;
         }
 
+        [SuppressMessage(
+            "StyleCop.CSharp.SpacingRules",
+            "SA1013:ClosingBracesMustBeSpacedCorrectly",
+            Justification = "Rule is incompatible with pattern matching in switch cases.")]
         private bool IsFatal(Exception ex)
         {
-            if (ex is StackOverflowException ||
-                ex is OutOfMemoryException)
-            {
-                return true;
-            }
-
-            if (ex is AggregateException aggregateException)
-            {
-                foreach (var childException in aggregateException.InnerExceptions)
-                {
-                    if (IsFatal(childException))
+            re: switch (ex)
+            {
+                case OutOfMemoryException _:
+                case StackOverflowException _:
+                    return true;
+                case AggregateException { InnerException: { }, // avoids allocation if collection is empty

```

Comment:
That's interesting. I didn't think that AggregateException would make sense without any InnerException. But apparently, the [implementation](https://referencesource.microsoft.com/#mscorlib/system/AggregateException.cs) allows for that. However, the .NET Source Code Browser shows a different implementation than the one you referenced. That one is always allocating (in the ctor). That would suggest that the allocation already happened anyway?

Reply:
btw: even with the version you pasted, I have a hard time finding a path, that would actually not allocate as part of the constructor. You could still say, it's more defensive.

Reply:
> However, the .NET Source Code Browser shows a different implementation than the one you referenced.

You are looking at the [.NET Framework source browser](https://referencesource.microsoft.com/#mscorlib/system/AggregateException.cs). You want to go over [source.dot.net](https://source.dot.net/) instead, which reflects the .NET 5 source code at this point in time.

> I didn't think that AggregateException would make sense without any InnerException.

Yes, you are right. No one in their RightMind&trade; 🤪 should be doing so (like `throw new AggregateException("Multiple errors.")`), but I've seen plenty of abuses in real life where people would use `AggregateException` for application errors and, for example, tuck away validation errors in the [`Data`](https://docs.microsoft.com/en-us/dotnet/api/system.exception.data?view=net-5.0) dictionary instead of defining a separate exception type. Other times, when they would have a separate exception type, I've seen it as a subclass of `AggregateException` (for whatever faulty reasoning). I guess that's made me a little paranoid over time but this was a cheap additional pattern to throw in so I did.

Reply:
In any case, the [`StackOverflowException` check isn't needed](https://github.com/fbeltrao/web-api-on-az-func/pull/7) so the only fatal exception being considered here now is `OutOfMemoryException`.

Reply:
@atifaziz thanks for your tips, learned a few things 😃 !